internal/services/tenant_isolation_policy/resource_tenant_isolation_policy_test.go

```diff
@@ -0,0 +1,605 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package tenant_isolation_policy_test
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/jarcoal/httpmock"
+	"github.com/microsoft/terraform-provider-power-platform/internal/mocks"
+)
+
+const testTenantID = "00000000-0000-0000-0000-000000000001"
+
+func setupTenantHttpMocks() {
+	// Mock tenant endpoint that's called before CRUD operations.
+	httpmock.RegisterResponder("GET", "https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/tenant?api-version=2021-04-01",
+		func(req *http.Request) (*http.Response, error) {
+			return httpmock.NewStringResponse(http.StatusOK, fmt.Sprintf(`{
+                "id": "/providers/Microsoft.BusinessAppPlatform/tenant",
+                "name": "default",
+                "type": "Microsoft.BusinessAppPlatform/tenant",
+                "tenantId": "%s",
+                "state": "Enabled"
+            }`, testTenantID)), nil
+		})
+
+	// Mock GET tenant isolation policy endpoint with empty policy.
+	httpmock.RegisterResponder("GET", fmt.Sprintf("https://api.bap.microsoft.com/providers/PowerPlatform.Governance/v1/tenants/%s/tenantIsolationPolicy", testTenantID),
+		func(req *http.Request) (*http.Response, error) {
+			return httpmock.NewStringResponse(http.StatusOK, `{
+                "properties": {
+                    "isDisabled": false,
+                    "allowedTenants": []
+                }
+            }`), nil
+		})
+}
+
+// TestAccTenantIsolationPolicy_Validate_Create tests the basic creation and import of a tenant isolation policy.
+// It verifies:
+// 1. Creation of a policy with a single allowed tenant (inbound and outbound access)
+// 2. The ability to import an existing policy and verify its attributes match the configuration.
+func TestAccTenantIsolationPolicy_Validate_Create(t *testing.T) {
```

Comment:
Import TestAcc should be called TestAccTenantIsolationPolicy_Validate_Import.

Reply:
I like that, we should have that for all the resources

Reply:
import test should only be done with unit tests since we can't rely on external infra existing.